### PR TITLE
feat: `ivy.experimental.tensor_train` #26583

### PR DESCRIPTION
**Pull request by mobley-trent**
This PR adds the `tensor_train` function to the ivy experimental API

Ref:
- [TensorLy](https://github.com/tensorly/tensorly/blob/main/tensorly/decomposition/_tt.py#L8)

### CODE REVIEW

File: ivy/functional/ivy/experimental/linear_algebra.py [hunk 1]
```diff
@@ -1153,6 +1153,86 @@ def truncated_svd(
...
+@handle_array_function
+@handle_device_shifting
+def tensor_train(
+    input_tensor: Union[ivy.Array, ivy.NativeArray],
+    rank: Union[int, Sequence[int]],
+    /,
+    *,
+    svd: Optional[Literal["truncated_svd"]] = "truncated_svd",
+    verbose: Optional[bool] = False,
+):
```
Comment (hello-fri-end) [inline ivy/functional/ivy/experimental/linear_algebra.py:1169]: let's add a type hint for the returned type here.

File: ivy/data_classes/container/experimental/linear_algebra.py [hunk 2]
```diff
@@ -1296,6 +1296,86 @@ def make_svd_non_negative(
...
+        rank: Union[Sequence[int], ivy.Container],
+        /,
+        *,
+        svd: Optional[Union[Literal["truncated_svd"], ivy.Container]] = "truncated_svd",
+        verbose: Optional[Union[bool, ivy.Container]] = False,
+        key_chains: Optional[Union[List[str], Dict[str, str], ivy.Container]] = None,
+        to_apply: Union[bool, ivy.Container] = True,
+        prune_unapplied: Union[bool, ivy.Container] = False,
+        map_sequences: Union[bool, ivy.Container] = False,
+    ):
```
Comment (hello-fri-end) [inline ivy/data_classes/container/experimental/linear_algebra.py:1351]: type-hint missing

File: ivy/data_classes/container/experimental/linear_algebra.py [hunk 3]
```diff
@@ -1296,6 +1296,86 @@ def make_svd_non_negative(
...
```
Comment (hello-fri-end) [inline ivy/data_classes/container/experimental/linear_algebra.py:1311]: type-hint mmissing.

File: ivy/data_classes/array/experimental/linear_algebra.py [hunk 4]
```diff
@@ -411,6 +411,37 @@ def make_svd_non_negative(
         """
         return ivy.make_svd_non_negative(self._data, U, S, V, nntype=nntype)
 
+    def tensor_train(
+        self: Union[ivy.Array, ivy.NativeArray],
+        rank: Union[int, Sequence[int]],
+        /,
+        svd: Optional[Literal["truncated_svd"]] = "truncated_svd",
+        verbose: Optional[bool] = False,
+    ):
```
Comment (hello-fri-end) [inline ivy/data_classes/array/experimental/linear_algebra.py:420]: type-hint missing.

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @mobley-trent , thanks for working on this. Just left a few minor comments. Happy to review again after you have resolved these :smile:

**Comment by mobley-trent** [comment]
@hello-fri-end I don't know why the ivy test is failing. Maybe because it can't parse the returned ivy.TTTensor data structure. But I think my implementation should've worked 🤔 
The tests I implemented from Tensorly are passing though

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py [hunk 5]
```diff
@@ -636,6 +636,30 @@ def _partial_tucker_data(draw):
...
+            max_num_dims=5,
+            min_dim_size=2,
+            max_dim_size=5,
+            min_value=0.1,
+            max_value=10.0,
+            ret_shape=True,
+        )
+    )
+    dims = len(shape)
+    rank = [1]
```
[the remark this answers is not in the document]
Comment (hello-fri-end) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py:690]: Hey @mobley-trent, can you explain this?

**Comment by hello-fri-end** [comment]
Could you also link the issue no. in the PR title please?

**Pull request merged**